pyccl/core.py

```diff
@@ -747,7 +758,31 @@ def compute_linear_power(self):
             rescale_mg = False
             pk = get_isitgr_pk_lin(self)
         elif trf == 'boltzmann_camb':
-            pk = get_camb_pk_lin(self)
+            pk_nl_from_camb = False
+            if self._config_init_kwargs['matter_power_spectrum'] == "camb":
+                pk_nl_from_camb = True
+            pk = get_camb_pk_lin(self, nonlin=pk_nl_from_camb)
+            if pk_nl_from_camb:
+                pk, pk_nl = pk
+                self._pk_nl['delta_matter:delta_matter'] = pk_nl
+                self._has_pk_nl = True
+                rescale_mg = False
+                rescale_s8 = False
+                if abs(self["mu_0"]) > 1e-14:
+                    warnings.warn("You want to compute the non-linear power "
```

Comment:
Humm, is this right? I think we didn't have this warning before, right?

Reply:
This is usually checked in `compute_nonlin_power` (https://github.com/LSSTDESC/CCL/blob/master/pyccl/core.py#L817) but since `has_nonlin_power` is set to `True`, that check needs to be done earlier.

Reply:
Sorry, you're right. I read too quickly and thought this was about neutrinos.